consensus/babe/babe.go

```diff
@@ -271,8 +284,14 @@ func (b *Session) buildBlock(parent *types.Header, slot Slot) (*types.Block, err
 		return nil, err
 	}
 
-	// initialize block
-	encodedHeader, err := scale.Encode(parent)
+	// create new block header
+	header, err := types.NewHeader(parent.Hash(), big.NewInt(0).Add(parent.Number, big.NewInt(1)), common.Hash{}, common.Hash{}, [][]byte{})
```

Comment:
I feel like we should break this up a bit for readability.

Reply:
which part? the block number?

Reply:
yes, block number would be nice to extract

Reply:
done